### DISCUSSION TITLE
Let `list` values in `tar_map()` keep names

### DISCUSSION THREAD

**Discussion by psychelzh** [ropensci/tarchetypes#90] [General]
The following codes generated a target from `tar_map()` to extract names from its `y` values, but in vain. It only returns `NULL`. Can the named list keep its names?

``` r
targets::tar_dir({
  targets::tar_script({
    library(targets)
    library(tarchetypes)
    values <- tibble::tribble(
      ~x, ~y,
      "X", list(a = 1)
    )
    tar_option_set(package = "tidyverse")
    targets <- tar_map(
      values = values,
      names = x,
      tar_target(
        test,
        names(y)
      )
    )
    list(targets)
  })
  targets::tar_make()
  targets::tar_read(test_X)
})
#> * start target test_X
#> * built target test_X
#> * end pipeline: 0.79 seconds
#> NULL
```

<sup>Created on 2022-03-31 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Ideally, the above target `test_X` will return `"a"` instead.

**Comment by psychelzh** [comment]
A second test shows that all the names are gone (even in the nested list)! So how to deal with this? Is there a workaround?

``` r
targets::tar_dir({
  targets::tar_script({
    library(targets)
    library(tarchetypes)
    values <- tibble::tribble(
      ~x, ~y,
      "X", list(a = list(b = 1))
    )
    tar_option_set(package = "tidyverse")
    targets <- tar_map(
      values = values,
      names = x,
      tar_target(
        test,
        y
      )
    )
    list(targets)
  })
  targets::tar_make()
  targets::tar_read(test_X)
})
#> * start target test_X
#> * built target test_X
#> * end pipeline: 0.8 seconds
#> [[1]]
#> [[1]][[1]]
#> [1] 1
```

<sup>Created on 2022-03-31 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by wlandau** [comment]
`tar_map()` works on [language objects](https://adv-r.hadley.nz/expressions.html), so it should work if you write `quote(list(a = list(b = 1)))` instead of `list(a = list(b = 1))`.

**Reply by psychelzh** [reply]
Thanks! It works! But should this behavior be in `tar_target_raw()`? Now I have no idea how to do this `quote()`s programmatically, i.e., generating a list of quoted values from existing `data.frame` values.

**Reply by psychelzh** [reply]
Eventually made it work by combining `rlang::expr_text()` and `rlang::parse_expr()`. Thank you very much!
